feat: geospatial compute endpoint

BenChand opened the pull request:
* Partially stabilize the new geospatial compute endpoint

saagar-cognite commented on packages/stable/src/api/geospatial/types.ts:
```diff
@@ -269,3 +269,38 @@ export interface GeospatialCRSResponse
+    };
+
+export interface GeospatialGeometryTransformComputeFunction {
+  geospatialGeometryComputeFunction: GeospatialGeometryComputeFunction;
```
suggestion(blocking): I think the key should be just `geometry`.

BenChand replied:
Oh you're right, will update

saagar-cognite commented on packages/stable/src/api/geospatial/types.ts:
```diff
@@ -269,3 +269,38 @@ export interface GeospatialCRSResponse
+
+export type GeospatialGeometryComputeFunction =
+  | {
+      geometryValue: GeospatialGeometryValueComputeFunction;
```
suggestion(blocking): the key here should be `ewkt`

BenChand replied:
That has been replaced with an interface

saagar-cognite commented on packages/stable/src/api/geospatial/types.ts:
```diff
@@ -269,3 +269,38 @@ export interface GeospatialCRSResponse
+  srid: number;
+}
+
+export type GeospatialGeometryComputeFunction =
```
suggestion(blocking): `GeospatialGeometryComputeFunction` should be just extending `GeospatialGeometryValueComputeFunction`.
Something like

```suggestion
      export interface GeospatialGeometryComputeFunction extends GeospatialGeometryValueComputeFunction {
      }
```

BenChand replied:
I think this has to be the other way around right? Because the `GeospatialGeometryValueComputeFunction` is just one function in the `GeospatialGeometryComputeFunction` group? Or at least it will be when the rest of the compute endpoint is stabilized

saagar-cognite commented on packages/stable/src/api/geospatial/types.ts:
```diff
@@ -269,3 +269,38 @@ export interface GeospatialCRSResponse
+      geometryValue: GeospatialGeometryValueComputeFunction;
+    };
+
+export interface GeospatialGeometryValueComputeFunction {
```
```suggestion
export interface GeospatialExtendedWellKnownText {
```

saagar-cognite replied:
should replace `GeospatialGemetryValueComputeFunction` in above functions as well.

saagar-cognite commented on packages/stable/src/api/geospatial/types.ts:
```diff
@@ -269,3 +269,38 @@ export interface GeospatialCRSResponse
+  items: GeospatialComputedItem[];
+}
+
+export interface GeospatialComputeRequest {
```
```suggestion
export interface GeospatialComputeRequest extends GeospatialJsonComputeOutput {
}
```

saagar-cognite replied:
```
interface GeospatialJsonComputeOutput {
output: Record<string, GeospatialComputeFunction>
}
```

saagar-cognite replied:
to make it in synergy with the docs

saagar-cognite commented on packages/stable/src/api/geospatial/types.ts:
```diff
@@ -269,3 +269,38 @@ export interface GeospatialCRSResponse
+}
+
+export interface GeospatialComputeRequest {
+  output: Map<string, GeospatialGeometryComputeFunction>
```
suggestion(blocking): `Map` should be replaced with `Record`

saagar-cognite commented on packages/stable/src/api/geospatial/types.ts:
```diff
@@ -269,3 +269,38 @@ export interface GeospatialCRSResponse
+}
+
+export interface GeospatialComputeResponse {
+  items: GeospatialComputedItemList;
```
suggestion(blocking): we generally do not need `items` in our types as that adapter is taken care by the base sdk.
Need to figure out which endpoint method we've to use for compute.
We've options like `listEndpoint`, `retrieveEndpoint`, etc.

saagar-cognite commented on packages/stable/src/api/geospatial/computeAPI.ts:
```diff
@@ -0,0 +1,21 @@
+   * [Compute custom json output structures or well known binary format responses based on calculation or selection of feature properties or direct values given in the request.](https://docs.cognite.com/api/v1/#operation/compute)
+   *
+   */
+  public compute = async (
```
suggestion(blocking): could you please check the implementation of `search` or `retrieve` features and see if we can use an existing `endpoint` method for not having to use list in our types and having the response already adapter to our need with `items` already extracted.
This is pretty standard in the sdk.

BenChand replied:
I looked through the various methods in the `baseResourceApi.ts` file and couldn't find an existing endpoint method that supported:

* No suffix to url
* No requirement of the query being a filter type
* Doesn't require the response to be a cursor list

Might be missing something, but seems to me like the custom response interface might be needed here (?)

BenChand committed:
feat(geospatial): add compute

commit 9261c0e321c21b572db740eb604c6fc3d424cba8
Merge: 07162a2b25 f94d6007f9
Author: Ben Chand <benchand1995@gmail.com>
Date:   Tue Nov 15 10:36:56 2022 +0100

    Merge branch 'master' into ben/geospatial-compute-endpoint

commit 07162a2b25b808d1edb023b5ddd8a5791f91439f
Author: Ben Chand <benchand1995@gmail.com>
Date:   Thu Oct 20 15:34:20 2022 +0000

    feat: restructuring geospatial compute types

commit 48dc281b3a5d6833ef99865683d39c3b98f0aa13
Author: Ben Chand <benchand1995@gmail.com>
Date:   Wed Oct 19 10:37:00 2022 +0200

    Rename property

commit 03be1b7c8eb5f1089983dab6237b5152088285d7
Author: Ben Chand <benchand1995@gmail.com>
Date:   Tue Oct 18 11:00:20 2022 +0000

    feat: geospatial compute endpoint

BenChand committed:
feat(geospatial): lint

BenChand committed:
feat(geospatial): test

BenChand committed:
feat(geospatial): fix nested list

BenChand committed:
feat(geospatial): remove a wrap

BenChand committed:
feat(geospatial): int!

BenChand committed:
feat(geospatial): some more linting

as a treat

BenChand committed:
feat(geospatial): too much linting go back

BenChand committed:
feat(geospatial): snippet

BenChand committed:
feat(geospatial): lints

lints for the lint god!

BenChand committed:
feat(geospatial): bit more linting

BenChand committed:
feat(geospatial): delete a newline

cognite-bulldozer[bot] committed:
Merge refs/heads/master into ben/geospatial-compute-endpoint

cognite-bulldozer[bot] committed:
Merge refs/heads/master into ben/geospatial-compute-endpoint

cognite-bulldozer[bot] committed:
Merge refs/heads/master into ben/geospatial-compute-endpoint

cognite-bulldozer[bot] committed:
Merge refs/heads/master into ben/geospatial-compute-endpoint

cognite-bulldozer[bot] committed:
Merge refs/heads/master into ben/geospatial-compute-endpoint

bdimitrijoski reviewed (commented):
@BenChand can you fix the tests please?

BenChand committed:
feat(geospatial): poke
